### PR TITLE
New GA version changes - 2021-10-01

### PR DESCRIPTION
**Pull request by sirireddy12**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new API version - 2021-10-01
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month. - October
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month. - September 9th
  4. If updating an existing version, please select the specific langauge SDKs and CLIs that must be refreshed after the swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No refresh required for updates in this PR

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following apply to the PR so that label “WaitForARMFeedback” will be added automatically to begin ARM API Review. Failure to comply may result in delays to the manifest.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits.

- [ ] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [ ] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by msyyc** [comment]
approve for python breaking change

File: specification/hybridkubernetes/resource-manager/readme.md [hunk 1]
```diff
@@ -28,7 +28,7 @@ These are the global settings for the Kubernetes Connect RP.
 ```yaml
 openapi-type: arm
 openapi-subtype: rpaas
-tag: package-2021-04-01-preview
+tag: package-2021-10-01
```
Comment (msyyc) [inline specification/hybridkubernetes/resource-manager/readme.md:31]: default tag should be changed to new version according to rules.

File: specification/hybridkubernetes/resource-manager/Microsoft.Kubernetes/stable/2021-10-01/examples/UpdateClusterExample.json [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    "api-version": "2021-10-01",
+    "subscriptionId": "1bfbb5d0-917e-4346-9026-1d3b344417f5",
+    "resourceGroupName": "k8sc-rg",
+    "clusterName": "testCluster",
+    "ConnectedClusterPatch": {
+      "tags": {
+        "tag1": "value1",
+        "tag2": "value2"
+      },
+      "properties": {
```
Comment (akashkeshari) [inline specification/hybridkubernetes/resource-manager/Microsoft.Kubernetes/stable/2021-10-01/examples/UpdateClusterExample.json:12]: Let's remove this property bag from the example. We wil actually be throwing error in this case

**Review by rapatchi — APPROVED** [review]

File: specification/hybridkubernetes/resource-manager/Microsoft.Kubernetes/stable/2021-10-01/examples/UpdateClusterExample.json (hunk 2 above)
Comment (sirireddy12) [reply specification/hybridkubernetes/resource-manager/Microsoft.Kubernetes/stable/2021-10-01/examples/UpdateClusterExample.json]: Removed it

File: specification/hybridkubernetes/resource-manager/Microsoft.Kubernetes/stable/2021-10-01/examples/ConnectedClustersListClusterCredentialResult.json [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    "api-version": "2021-10-01",
+    "Properties": {
+      "authenticationMethod": "AAD",
+      "clientProxy": true
+    }
+  },
+  "responses": {
+    "200": {
+      "body": {
+        "kubeconfigs": [
```
Comment (krdhruva) [inline specification/hybridkubernetes/resource-manager/Microsoft.Kubernetes/stable/2021-10-01/examples/ConnectedClustersListClusterCredentialResult.json:15]: Should this have hybrid config in response?

File: specification/hybridkubernetes/resource-manager/Microsoft.Kubernetes/stable/2021-10-01/examples/ConnectedClustersListClusterCredentialResultNonAad.json [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    "api-version": "2021-10-01",
+    "Properties": {
+      "authenticationMethod": "Token",
+      "clientProxy": true
+    }
+  },
+  "responses": {
+    "200": {
+      "body": {
+        "kubeconfigs": [
```
Comment (krdhruva) [inline specification/hybridkubernetes/resource-manager/Microsoft.Kubernetes/stable/2021-10-01/examples/ConnectedClustersListClusterCredentialResultNonAad.json:15]: same as before: HybridConnectionConfig object should be there where CSP flag is true. We should have one example without CSP flag also.

**Review by krdhruva — CHANGES_REQUESTED** [review]
🕐

File: specification/hybridkubernetes/resource-manager/Microsoft.Kubernetes/stable/2021-10-01/examples/ConnectedClustersListClusterCredentialResult.json (hunk 3 above)
Comment (sirireddy12) [reply specification/hybridkubernetes/resource-manager/Microsoft.Kubernetes/stable/2021-10-01/examples/ConnectedClustersListClusterCredentialResult.json:15]: Added it

File: specification/hybridkubernetes/resource-manager/Microsoft.Kubernetes/stable/2021-10-01/examples/ConnectedClustersListClusterCredentialResultNonAad.json (hunk 4 above)
Comment (sirireddy12) [reply specification/hybridkubernetes/resource-manager/Microsoft.Kubernetes/stable/2021-10-01/examples/ConnectedClustersListClusterCredentialResultNonAad.json:15]: Added it

**Comment by rapatchi** [comment]
>       "clientProxy": true

this should be false

---
Refers to: specification/hybridkubernetes/resource-manager/Microsoft.Kubernetes/preview/2021-04-01-preview/examples/ConnectedClustersListClusterCredentialsResult.json:9 in 3c5ef91. [](commit_id = 3c5ef91b2f2141edd75d03471e03ae7e20a4b8b9, deletion_comment = True)

**Comment by rapatchi** [comment]
> {

The file names are little confusing.
Could you rename them to some consistent format?.

1. ....CSPAAD.json
2. ....CSPToken.json
3. .....HPAAD.json
4. .....HPToken.json ?.


---
Refers to: specification/hybridkubernetes/resource-manager/Microsoft.Kubernetes/preview/2021-04-01-preview/examples/ConnectedClustersListClusterCredentialsResult.json:1 in 3c5ef91. [](commit_id = 3c5ef91b2f2141edd75d03471e03ae7e20a4b8b9, deletion_comment = True)

File: specification/hybridkubernetes/resource-manager/Microsoft.Kubernetes/stable/2021-10-01/examples/ConnectedClustersListClusterCredentialResult.json (hunk 3 above)
Comment (rapatchi) [inline specification/hybridkubernetes/resource-manager/Microsoft.Kubernetes/stable/2021-10-01/examples/ConnectedClustersListClusterCredentialResult.json:15]: Don't see HC details in the kubeconfig,

**Review by rapatchi — APPROVED** [review]

**Review by krdhruva — APPROVED** [review]

**Comment by majastrz** [comment]
It doesn't look like you structured your history to make it easy to review the delta for us. These were the instructions in the PR title:
image

Can you also link to the previously approved API so I can review the delta?

**Comment by rapatchi** [comment]
> It doesn't look like you structured your history to make it easy to review the delta for us. These were the instructions in the PR title:
> image
> 
> Can you also link to the previously approved API so I can review the delta?

We have taken the current GA api-version as base with no change from: azure-rest-api-specs/connectedClusters.json at main · Azure/azure-rest-api-specs (github.com)
And as we want to take “listClusterUserCredentials” defined in: azure-rest-api-specs/connectedClusters.json at main · Azure/azure-rest-api-specs (github.com)
To GA.
The only change which we did while taking this API to GA is rename it from “listClusterUserCredentials” to “listClusterUserCredential”. As it should have been that in the first place.

Hope this helps.

**Review by majastrz — APPROVED** [review]
Signed off from ARM side.

**Comment by majastrz** [comment]
>
> Hope this helps.

Yes, it does. Thanks!

**Pull request merged**